### ISSUE TITLE
Regarding training of yolov8 model through  git clone locally ?

### ISSUE THREAD

**Issue by wolverinex24** [ultralytics/ultralytics#5796]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

how to start training the yolov8 classify model after git cloning locally and adjusting the default.yaml ?
Actually, I want to experiment with other backbone and I didn't get step-wise instructions regarding starting  yolov8 locally from cloned  git

**Comment by glenn-jocher** [comment]
@wolverinex24 thank you for your question and interest in working with YOLOv8.

Once you have cloned the YOLOv8 repo locally, the first step to conduct training is to prepare your dataset. Remember to structure your dataset in the required format, with separate directories for training and validation images.

After preparing your dataset, you need to define your model configuration and experiment with different backbones. You'd do this in the 'models' folder. Here, you can modify the 'default.yaml' file or create a new YAML file for your model's configuration. You can specify the type of backbone you want to use and make other adjustments as necessary.

When you've set up your model configuration file, you can now train your model. This is done via the command line in your local environment. You'll need to run the 'train.py' script and specify your model configuration file and the path to your data with the appropriate flags.

As you proceed, please make sure you comply with the requirements for running this model, such as having the correct version of PyTorch installed in your local environment. I hope this helps, and please don't hesitate to ask if you have more questions!

**Comment by wolverinex24** [comment]
I have made all the changes in default.yaml file now what should be complete command line instruction to run train,py file using my default.yaml. if there is any documentation to which I can refer (just made a few changes in default.yaml file  and currently not trying with the backbone changed )
thank you for giving your precious time!!

**Comment by glenn-jocher** [comment]
I'm glad to hear you've made progress with your setup, @wolverinex24! To train your YOLOv8 model with your customized `default.yaml` file, you will execute the `train.py` script from your command line. Here is a general overview of the command you should use, without going into specific flags or parameters:

```bash
python path/to/train.py --data path/to/your_dataset.yaml --cfg path/to/default.yaml --weights '' --batch-size 16 --epochs 50
```

In this command, you will need to replace:

- `path/to/train.py` with the path to the `train.py` file within your cloned YOLOv8 directory.
- `path/to/your_dataset.yaml` with the path to a YAML file that describes your dataset's configuration. This file should contain paths to your training and validation data, as well as class names and number of classes.
- `path/to/default.yaml` with the path to your customized `default.yaml` file unless it is already in the expected directory by `train.py`.
- The value for `--weights` should generally be empty for training from scratch, but if you're using pre-trained weights, specify the path here.
- `--batch-size` and `--epochs` are set to example values. You should set these based on your dataset size and hardware capabilities.

For detailed instructions, error troubleshooting, and additional flags you can utilize during training, I recommend referring to our official documentation at Ultralytics Docs. The documentation will give you a complete understanding of all the available options for training YOLOv8 models, as well as best practices for achieving good results.

Please follow the information provided into our official documentation for a better understanding, and reach out again if you encounter any issues or have further questions. Happy training!

**Comment by glenn-jocher** [comment]
Hello @N786h,

Thank you for your question! After you execute the training command, the results will be saved in a directory named `runs/train/exp` by default. This directory will contain several important files and subdirectories, including:

1. **`weights/`**: This folder contains the model weights saved at different stages of training, including `best.pt` and `last.pt`.
2. **`results.png`**: A plot of training and validation metrics over epochs, such as loss and mAP.
3. **`events.out.tfevents...`**: TensorBoard logs that you can visualize using TensorBoard to monitor training metrics in real-time.
4. **`train_batch0.jpg`, `val_batch0_labels.jpg`, etc.**: Sample images with predictions and labels for quick visual inspection.

To visualize the training progress, you can use TensorBoard. Simply run the following command in your terminal:

```bash
tensorboard --logdir runs/train/exp
```

Then, open your web browser and navigate to `http://localhost:6006/` to see the training metrics and visualizations.

If you have any further questions or run into any issues, please feel free to ask. We're here to help! 😊
